src/fdb5/toc/TocStore.cc

```diff
@@ -264,9 +266,8 @@ eckit::PathName TocStore::getDataPath(const Key& key) const {
 }
 
 void TocStore::flushDataHandles() {
-
-    for (HandleStore::iterator j = handles_.begin(); j != handles_.end(); ++j) {
-        eckit::DataHandle *dh = j->second;
+    std::lock_guard<std::recursive_mutex> lock(handlesMutex_);
+    for (const auto& [p, dh] : handles_) {
```

Comment:
I like it

Reply:
`std::lock_guard lock(handlesMutex_);` would work (for all those guards; no need for explicit type)
also, including explicitly `<mutex>` header as a good practice